### PR TITLE
[Refactor] refactor RotLocalVisualizer

### PR DESCRIPTION
**Pull request by zytx121**
## Reviewer
@chhluo @hhaAndroid

### CODE REVIEW

File: mmrotate/visualization/local_visualizer.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import List, Optional
+
+import numpy as np
+import torch
+from mmdet.structures.mask import bitmap_to_polygon
+from mmdet.visualization import DetLocalVisualizer
+from mmdet.visualization.palette import _get_adaptive_scales, get_palette
```
Comment (jbwang1997) [inline mmrotate/visualization/local_visualizer.py:8]: get_palette may be moved into local_visualizer.py. It's better to directly import from mmdet.visualization.

File: tests/test_visualization/test_local_visualizer.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        rot_local_visualizer.add_datasample(
+            'image',
+            image,
+            gt_rot_data_sample,
+            pred_rot_data_sample,
+            draw_pred=False,
+            out_file=out_file)
+        self._assert_image_and_shape(out_file, (h, w, 3))
+
+        # TODO: test gt_panoptic_seg
```
Comment (chhluo) [inline tests/test_visualization/test_local_visualizer.py:83]: remove it

File: mmrotate/visualization/local_visualizer.py (hunk 1 above)
Comment (chhluo) [reply mmrotate/visualization/local_visualizer.py:8]: see https://github.com/open-mmlab/mmdetection/pull/8517

File: tests/test_visualization/test_local_visualizer.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        num_class = 3
+        num_bboxes = 5
+        out_file = 'out_file.jpg'
+
+        image = np.random.randint(0, 256, size=(h, w, 3)).astype('uint8')
+
+        # test gt_instances
+        gt_instances = InstanceData()
+        gt_instances.bboxes = _rand_rbboxes(num_bboxes, h, w)
+        gt_instances.labels = torch.randint(0, num_class, (5, ))
```
Comment (chhluo) [inline tests/test_visualization/test_local_visualizer.py:34]: `5` -> `num_bboxes`

File: tests/test_visualization/test_local_visualizer.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            'image', image, gt_rot_data_sample, out_file=out_file)
+        assert os.path.exists(out_file)
+        drawn_img = cv2.imread(out_file)
+        assert drawn_img.shape == (h, w, 3)
+        os.remove(out_file)
+
+        # test gt_instances and pred_instances
+        pred_instances = InstanceData()
+        pred_instances.bboxes = _rand_rbboxes(num_bboxes, h, w)
+        pred_instances.labels = torch.randint(0, num_class, (5, ))
```
Comment (chhluo) [inline tests/test_visualization/test_local_visualizer.py:52]: `5` -> `num_bboxes`

File: tests/test_visualization/test_local_visualizer.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        pred_instances.scores = torch.rand((5, ))
```
Comment (chhluo) [inline tests/test_visualization/test_local_visualizer.py:53]: same as above

**Comment by jbwang1997** [comment]
Suggest to support both rbox and qbox.

File: mmrotate/visualization/local_visualizer.py (hunk 1 above)
Comment (zytx121) [reply mmrotate/visualization/local_visualizer.py:8]: Waiting for https://github.com/open-mmlab/mmdetection/pull/8517 to be merged.

**Comment by zytx121** [comment]
> Suggest to support both rbox and qbox.

Good suggestion! We will do it after boxlist is fully merged.

> Suggest to support both rbox and qbox.

In dev-1.x, all models output will be qbox. This PR will modify and no hurry to merge.

File: mmrotate/visualization/local_visualizer.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import List, Optional
+
+import numpy as np
+import torch
+from mmdet.structures.mask import bitmap_to_polygon
+from mmdet.visualization import DetLocalVisualizer, get_palette
+from mmdet.visualization.palette import _get_adaptive_scales
```
Comment (chhluo) [inline mmrotate/visualization/local_visualizer.py:8]: `from mmdet.visualization import _get_adaptive_scales`

mmdet.visualization.palette will be deleted.

File: mmrotate/visualization/local_visualizer.py [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        text_color (str, tuple(int), optional): Color of texts.
+            The tuple of color should be in BGR order.
+            Defaults to (200, 200, 200).
+        mask_color (str, tuple(int), optional): Color of masks.
+            The tuple of color should be in BGR order.
+            Defaults to None.
+        line_width (int, float): The linewidth of lines.
+            Defaults to 3.
+        alpha (int, float): The transparency of bboxes or mask.
+                Defaults to 0.8.
```
Comment (chhluo) [inline mmrotate/visualization/local_visualizer.py:38]: indentation

File: tools/misc/browse_dataset.py [hunk 8]
```diff
@@ -45,63 +39,45 @@ def parse_args():
...
+    # register all modules in mmrotate into the registries
+    register_all_modules()
 
-    progress_bar = mmcv.ProgressBar(len(dataset))
+    dataset = DATASETS.build(cfg.train_dataloader.dataset)
+
+    visualizer = VISUALIZERS.build(cfg.visualizer)
+    visualizer.dataset_meta = dataset.metainfo
 
+    progress_bar = mmcv.ProgressBar(len(dataset))
```
Comment (chhluo) [inline tools/misc/browse_dataset.py:56]: `ProgressBar` is moved to `mmengine.utils`, https://github.com/open-mmlab/mmdetection/pull/8594

**Review by aniyah899 — COMMENTED** [review]
hi

**Pull request closed without merging**